framework/platform/bpl/uci/cfg/bpl_cfg.cpp

```diff
@@ -175,33 +175,30 @@ int cfg_get_device_info(BPL_DEVICE_INFO *device_info) { return 0; }
 
 int cfg_get_wifi_params(const char iface[BPL_IFNAME_LEN], struct BPL_WLAN_PARAMS *wlan_params)
 {
-    int retVal = 0;
-    int index  = 0;
-
     if (!iface || !wlan_params) {
+        MAPF_ERR("cfg_get_wifi_params: invalid input: iface = "
+                 << intptr_t(iface) << " wlan_params = " << intptr_t(wlan_params));
         return RETURN_ERR;
     }
 
-    retVal = cfg_get_index_from_interface(iface, &index);
-    if (retVal) {
-        return retVal;
-    }
+    char ifname[BPL_IFNAME_LEN] = {0};
+    utils::copy_string(ifname, iface, BPL_IFNAME_LEN);
 
     // The UCI "disabled" setting is optional, defaults to false if not present
     bool disabled = false;
-    cfg_uci_get_wireless_bool(TYPE_RADIO, index, "disabled", &disabled);
+    cfg_uci_get_wireless_bool(TYPE_RADIO, ifname, "disabled", &disabled);
```

Comment:
Not OK - this can fail, I know this wasn't tested before but it should definitely be tested now since `ifname` is not guaranteed to be in `/etc/config/wireless`.

Reply:
Well... If it fails, it can be because the section is missing (a real error), or because the option is not set in UCI (default false, as explained in the comment above). So this *is* actually OK IMO.

Reply:
Right, I get it.